apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/wpcom-block-editor-nav-sidebar.tsx

```diff
@@ -0,0 +1,132 @@
+	title: { raw: string; rendered: string };
+}
+
+const Button = ( {
+	children,
+	...rest
+}: OriginalButton.Props & { icon?: any; iconSize?: number } ) => (
+	<OriginalButton { ...rest }>{ children }</OriginalButton>
+);
+
+export default function WpcomBlockEditorNavSidebar() {
+	const [ items, isOpen, postType ] = useSelect( ( select ) => {
+		const { getPostType } = select( 'core' ) as any;
+
+		return [
+			selectNavItems( select ),
+			select( STORE_KEY ).isSidebarOpened(),
+			getPostType( select( 'core/editor' ).getCurrentPostType() ),
+		];
+	} );
+
+	const prevIsOpen = useRef( isOpen );
+	const [ isClosing, setIsClosing ] = useState( false );
+
+	useEffect( () => {
+		if ( isOpen ) {
+			document.body.classList.add( 'is-wpcom-block-editor-nav-sidebar-close-hidden' );
+		} else if ( prevIsOpen.current ) {
+			// Check previous isOpen value so we don't set isClosing on first mount
+			setIsClosing( true );
+		}
+
+		prevIsOpen.current = isOpen;
+	}, [ isOpen, prevIsOpen, setIsClosing ] );
+
+	const { toggleSidebar } = useDispatch( STORE_KEY );
+
+	return (
+		<div className="wpcom-block-editor-nav-sidebar__container" aria-hidden={ ! isOpen }>
+			{ ( isOpen || isClosing ) && (
+				<div className="wpcom-block-editor-nav-sidebar__header">
+					<Button
+						icon={ wordpress }
+						iconSize={ 36 }
+						className={ classNames( {
+							'wpcom-block-editor-nav-sidebar__is-shrinking': isOpen,
+							'wpcom-block-editor-nav-sidebar__is-growing': isClosing,
+						} ) }
+						onClick={ () => {
+							if ( isOpen ) {
+								// The `useEffect` above already takes care of setting isClose to true,
+								// but there's a flicker where isOpen and isClosing are both false for
+								// a brief moment in time. Setting isClose to true here too to avoid
+								// the flicker.
+								setIsClosing( true );
+							}
+							toggleSidebar();
+						} }
+						onAnimationEnd={ ( ev: any ) => {
+							if ( ev.animationName === 'wpcom-block-editor-nav-sidebar__grow' ) {
+								setIsClosing( false );
+								document.body.classList.remove( 'is-wpcom-block-editor-nav-sidebar-close-hidden' );
+							}
+						} }
+					/>
+				</div>
+			) }
+			<div className="wpcom-block-editor-nav-sidebar__header-space" />
+			<div className="wpcom-block-editor-nav-sidebar__home-button-container">
+				<Button
+					href={ addQueryArgs( 'edit.php', {
+						post_type: postType.slug,
+					} ) }
+					className="wpcom-block-editor-nav-sidebar__home-button"
+					icon={ chevronLeft }
+				>
+					{ get( postType, [ 'labels', 'all_items' ], __( 'Back' ) ) }
+				</Button>
+			</div>
+			<ul className="wpcom-block-editor-nav-sidebar__page-list">
+				{ items.map( ( item ) => (
+					<NavItem key={ item.id } item={ item } />
+				) ) }
+			</ul>
+		</div>
+	);
+}
+
+interface NavItemProps {
+	item: Post;
+}
+
+function NavItem( { item }: NavItemProps ) {
+	return (
+		<li>
+			<div>{ item.title.rendered }</div>
+			<pre>{ `/${ item.slug }/` }</pre>
```

Comment:
Another fun CSS one 😄, for longer post slugs, using default `<pre` formatting, they'll overflow the sidebar container like so:

![image](https://user-images.githubusercontent.com/14988353/82533945-a1765b00-9b87-11ea-9c46-7a51cf09f498.png)

Then when you close the sidebar, they peek out from behind the edge of the page 😆

![image](https://user-images.githubusercontent.com/14988353/82534018-c4a10a80-9b87-11ea-87ef-b1b064c3ad16.png)

I think we can resolve this by adding a class to this `<pre>` and then setting the attribute [`white-space`](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space) to a value like `pre-wrap`.

Reply:
What do you think about fixing this in a follow up? The navigation item styling has a few things wrong according to the mockup, and I was planning on dealing with hover style, font, wrapping, draft badges, etc. in future PRs.

Reply:
Absolutely! I was actually in the process of writing the Approving review 😄, just a sec!